packages/@aws-cdk/aws-cognito/lib/user-pool-client.ts

```diff
@@ -320,7 +320,7 @@ export class UserPoolClient extends Resource implements IUserPoolClient {
       explicitAuthFlows: this.configureAuthFlows(props),
       allowedOAuthFlows: props.disableOAuth ? undefined : this.configureOAuthFlows(),
       allowedOAuthScopes: props.disableOAuth ? undefined : this.configureOAuthScopes(props.oAuth),
-      callbackUrLs: callbackUrls && callbackUrls.length > 0 ? callbackUrls : undefined,
+      callbackUrLs: callbackUrls && callbackUrls.length > 0 && !props.disableOAuth ? callbackUrls : undefined,
       logoutUrLs: props.oAuth?.logoutUrls,
```

Comment:
I believe we should apply this for logoutUrLs as well.

Reply:
I looked into it, but logoutUrls are embedded inside the `oAuth` (`OAuthSettings` struct) property.
There's validation that currently verifies that `oAuth` and `disableOAuth` are not both simultaneously defined.

It's why I ended up leaving it alone. Let me know if I'm misreading though!

Reply:
Oh yes. That's fine then.